File: app/src/main/java/doit/study/droid/data/Converters.java
package doit.study.droid.data;

import android.arch.persistence.room.TypeConverter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Converters {
    @TypeConverter
    public static List<String> fromString(String value) {
        return value == null ? null : splitItems(value);
    }

    @TypeConverter
    public static String toString(List<String> data) {
        return data == null ? null : listToString(data);
    }

    public static String listToString(List<String> data){
        StringBuilder sb = new StringBuilder();
        for(String d: data)
            sb.append(d).append("\n");
        return sb.toString();
    }

    private static List<String> splitItems(String s){
        if ("".equals(s))
            return new ArrayList<>();
        else
            return Arrays.asList(s.split("\n"));
    }
}

File: app/src/main/java/doit/study/droid/data/local/Converters.kt
package doit.study.droid.data.local

import androidx.room.TypeConverter
import com.google.gson.Gson

class Converters {
    @TypeConverter
    fun listToJson(value: List<String>?): String {
        return Gson().toJson(value)
    }

    @TypeConverter
    fun jsonToList(value: String): List<String>? {
        return Gson().fromJson(value, Array<String>::class.java).toList()
    }
}
